Agents/PubChemAgent/README.md

```diff
@@ -0,0 +1,195 @@
+```
+
+Time out issues have been observed when building the image. If this happens, please try pulling the required stack-clients image first by
+
+```bash
+docker pull docker.cmclinnovations.com/stack-client:1.6.2
+```
+
+### Docker deployment
+
+Deploy the dockerised agent by running the following code in the command prompt from the same location where this README is located (ideally, use a bash terminal to avoid potential issues with inconsistent path separators).
+
+```bash
+# Deploy the Docker images (production / debug) locally
+docker-compose -f docker-compose.yml  up
+docker-compose -f docker-compose.debug.yml  up
+```
+
+To verify the correct startup of the agent, open the URL address the agent is running on, e.g. <http://localhost:5000> in your browser.
+
+### Stack Deployment
+
+If you want to spin up this agent as part of a stack, do the following:
+
+- Build the production image using the commands provided above (do not spin up the image)
+- Copy the pubchem-agent.json file from the stack-manager-input-config folder into the inputs/config/services folder of the stack manager
+- Copy os_stack.json into the inputs/config folder of the stack manager
+- Start the stack manager (i.e. bash ./stack.sh start os_stack from the stack-manager repo). This should start the container. Please use a bash terminal to avoid potential issues with inconsistent path separators.
+- The agent shall become available at ```<http://<HOST>:<PORT>/>```
+
+### Notes on debugging
+
+The stack deployment of the agent is focused on the production image of the agent. To debug the agent, it is easiest to deploy the debug version locally by providing all required parameters in the docker-compose.debug.yml and running the below commands (build as required):
+
+```bash
+docker-compose -f docker-compose.debug.yml  build
+docker-compose -f docker-compose.debug.yml  up
+```
+
+This spins up the agent on <http://localhost:5000>, waiting for the debugger to attach. To attach to the container and start debugging, please use the provided Python: Debug Flask within Docker debug configuration. Although outside the stack, this procedure allows to debug all essential functionality of the agent (without the need to have a full stack running).
+
+## How to use
+
+The agent can be used as a simple command line tool or as a web agent.
+
+### Command line usage
+
+ The agent can be run from the command line via the `pubchemagent` command which accepts the following options:
+
+ ```bash
+Usage:
+    pubchemagent  (--inchi=<inchi>)
+
+Options:
+--inchi=<inchi>                     inchi string
+```
+
+Example usage:
+
+```bash
+(pubchemagent_venv) $ pubchemagent --inchi='InChI=1/Ar'
+```
+
+### Web agent usage
+
+The web agent uses the [TheWorldAvatar](https://github.com/cambridge-cares/TheWorldAvatar) project knowledge graph to retrieve most of the species data needed
+Only one input is required in the agent main route:
+
+These are:
+
+/pubchemagent/query?`inchi=InChI=inchistring`
```

Comment:
Does this need updating? It certainly didn't work for me. I had to look through the code to figure out that this works (within a stack):
`curl http://localhost:3838/pubchemagent/query/species?inchi='InChI=1/Ar'`
You may want to include that (or something like that) as a working example.

Reply:
I updated it